### PR TITLE
ACC-60: Compare Translation branches with correct base branch

### PR DESCRIPTION
**Pull request by mgreau**
The translation branches for released PLF versions are now compared with stable/X.Y.x instead of develop

   * update shell function clone_or_fetch_git_repo to add blessed remote on all cloned repositories
   * update PHP function getTranslationBranches to compare branches on exoplatform or exodev organization

### CODE REVIEW

File: _functions_git.sh [hunk 1]
```diff
@@ -36,13 +36,28 @@ clone_or_fetch_git_repo() {
...
+    pushd ${_src_dir}/${_repo}.git > /dev/null 2>&1
+    echo_info "Add blessed remote for exoplatform organization..."
+    git remote add blessed git@github.com:exoplatform/${_repo}.git
   else
     pushd ${_src_dir}/${_repo}.git > /dev/null 2>&1
     set +e
     status=0
     git remote set-url origin git@github.com:${_orga}/${_repo}.git
+    # Add remote named blessed for exoplatform organization
+    git ls-remote --exit-code blessed
```
Comment (vsellier) [inline _functions_git.sh]: Does the output should not be redirected to /dev/null also ?

File: _functions_git.sh [hunk 2]
```diff
@@ -54,7 +69,17 @@ clone_or_fetch_git_repo() {
       git clone -v git@github.com:/${_orga}/${_repo}.git ${_src_dir}/${_repo}.git
       echo_info "Clone done ..."
       pushd ${_src_dir}/${_repo}.git > /dev/null 2>&1
-      git fetch --progress --prune origin
+      set +e
+      status=0
+      # Add remote named blessed for exoplatform organization if it doesn't exist
+      git ls-remote --exit-code blessed
```
Comment (vsellier) [inline _functions_git.sh]: Does the output should not be redirected to /dev/null also ?

File: _functions_git.sh [hunk 3]
```diff
@@ -36,13 +36,28 @@ clone_or_fetch_git_repo() {
...
+    # Add remote named blessed for exoplatform organization
+    git ls-remote --exit-code blessed
+    status=$?
+    set -e
+    if [ ${status} -ne 0 ]; then
+      echo_info "Add blessed remote for exoplatform organization..."
+      git remote add blessed git@github.com:exoplatform/${_repo}.git
+    fi
+    set +e
+    status=0
```
Comment (vsellier) [inline _functions_git.sh]: I see it was on the old code, but this affectation is not needed

File: _functions_git.sh [hunk 4]
```diff
@@ -54,7 +69,17 @@ clone_or_fetch_git_repo() {
       git clone -v git@github.com:/${_orga}/${_repo}.git ${_src_dir}/${_repo}.git
       echo_info "Clone done ..."
       pushd ${_src_dir}/${_repo}.git > /dev/null 2>&1
-      git fetch --progress --prune origin
+      set +e
+      status=0
```
Comment (vsellier) [inline _functions_git.sh]: same as previously

File: var/www/lib/functions.php [hunk 5]
```diff
@@ -204,21 +204,29 @@ function getTranslationBranches($projects)
...
+          }
+          $github_http_integration_org = $matches[1];
         }
-        $features[$branch][$project]['http_url'] = "https://github.com/" . $github_org . "/" . $github_repo . "/tree/integration/" . $branch;
+        $features[$branch][$project]['http_url'] = "https://github.com/" . $github_http_integration_org . "/" . $github_repo . "/tree/integration/" . $branch;
         // Add link to GitHub diff URL
-        $features[$branch][$project]['http_url_behind'] = "https://github.com/" . $github_org . "/" . $github_repo . "/compare/integration/" . $branch."...develop";
-        $features[$branch][$project]['http_url_ahead'] = "https://github.com/" . $github_org . "/" . $github_repo . "/compare/develop" ."...integration/".$branch;
-        $behind_commits_logs = $repoObject->git("log origin/integration/" . $branch . "..origin/develop --oneline");
+        $features[$branch][$project]['http_url_behind'] = "https://github.com/" . $github_http_integration_org . "/" . $github_repo . "/compare/integration/" . $branch."..." . $github_org . ":" . $baseBranchToCompareWith ."";
```
Comment (vsellier) [inline var/www/lib/functions.php]: why the final ."" ?

File: var/www/lib/functions.php [hunk 6]
```diff
@@ -204,21 +204,29 @@ function getTranslationBranches($projects)
...
+        $features[$branch][$project]['http_url_ahead'] = "https://github.com/" . $github_org . "/" . $github_repo . "/compare/" . $baseBranchToCompareWith ."" ."..." . $github_http_integration_org . ":integration/".$branch;
```
Comment (vsellier) [inline var/www/lib/functions.php]: ."" ."..." seems overkill ;)

File: var/www/lib/functions.php (hunk 5 above)
Comment (mgreau) [inline var/www/lib/functions.php]: because of search and replace ;)
I fix it

File: var/www/lib/functions.php (hunk 6 above)
Comment (mgreau) [inline var/www/lib/functions.php]: yes, fixed :)

**Comment by vsellier** [comment]
Thanks, LGTM

**Pull request merged**